### DISCUSSION TITLE
[Question] Is ext_mesh_gpu_instancing performance any different to using the same mesh in multiple nodes?

### DISCUSSION THREAD

**Discussion by camnewnham** [atteneder/glTFast#382] [Ideas]
Looking at the [implementation](https://github.com/atteneder/glTFast/commit/a212539b3ecfe07d65014a563075504bca80cc70) it seems that this creates a GameObject for each transform provided by the extension.

Aside from perhaps more efficient parsing of the transformation arrays and some minor less file-size in the GLTF JSON component, once the geometry is instantiated is this functionally any different to referencing the same mesh/material combination in multiple nodes, and providing different Translation/Rotation/Scale on the nodes in classic gltf fashion?

**Comment by camnewnham** [comment]
From a bit of reading, it looks like the _ideal_ scenario here would be to use `DrawMeshInstancedIndirect`, but this is not an insignificant amount of work to implement as it has to be supported in the shader, though this would also affect other unity interactions with the object.

**Reply by atteneder** [reply]
You're right on the spot!

The current implementation is straightforward. Profiling showed that Unity does a fairly good job of batching those draw calls though.

As you said, doing proper GPU instancing involves a bit more research and work. That's why I didn't tackle it yet. I'd also love to look at how this will be handled with DOTS.

**Reply by camnewnham** [reply]
Agreed; the draw calls don't seem to be much different. The main issue is that 10,000 GameObjects starts to make things pretty slow. 
And yes - makes sense to investigate ECS as the mechanism for this.

**Comment by atteneder** [comment]
Another thought, not related to rendering performance:

Using the same mesh in lots of nodes slows down loading time, since the JSON will be bloated and thus the parsing time increases. Using the GPU instancing extension won't suffer in the same way.
